### PR TITLE
Update fixed horizon notebook to reflect changes in the method

### PR DESCRIPTION
**Pull request by ruifanp**

### CODE REVIEW

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 1]
```diff
@@ -22,21 +12,19 @@
     "\n",
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:12]: In Fig. 1 `20-day`

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 2]
```diff
@@ -22,21 +12,19 @@
...
     "The following paper outlines the technique in greater depth, and is followed closely in this notebook tutorial: [\"Classification-based Financial Markets Prediction using Deep Neural Networks\"](https://arxiv.org/abs/1603.08604) by _Dixon et al._ (2016). \n",
     "\n",
     "\n",
     "## How it works\n",
     "\n",
-    "Fixed time horizon is a common method used in labeling financial data, usually applied on time bars. The forward rate of return relative\n",
-    "to $t_0$ over time horizon $h$ is calculated as follows:\n",
-    "\n",
+    "Fixed time horizon is a common method used in labeling financial data, usually applied on time bars. The return relative to $t_0$ over a time horizon $h$ is calculated as follows, assuming returns are lagged to make them forward looking:\n",
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:22]: `forward-looking`

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 3]
```diff
@@ -51,19 +39,17 @@
...
-    "volume bars instead of time bars, using data sampled at the same time every day (e.g. closing prices) or inputting a dynamic threshold\n",
-    "as a pd.Series corresponding to the times in the dataset. Another solution is to apply the labels to data based on standardized returns \\[Lopez de Prado, 2020: p.66\\]. In this case, the data is scaled by mean and standard deviation based on a given rolling window before labels are applied.\n",
+    "To avoid overlapping return windows, rather than specifying $h$, the user is given the option of resampling the returns to get the desired return period. Optionally, returns can be standardized by scaling by the mean and standard deviation of a rolling window. If threshold is a pd.Series, **threshold.index and prices.index must match**; otherwise labels will fail to be returned. If resampling is used, the threshold must match the index of prices after resampling. This is to avoid the user being forced to fill in meaningless thresholds. Acceptable strings inputs for resampling can be found [here](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#dateoffset-objects).\n",
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:42]: Same comment to `meaningless thresholds` as in [PR#381](https://github.com/hudson-and-thames/mlfinlab/pull/381/files)

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 4]
```diff
@@ -221,34 +206,41 @@
...
     "msft_returns = close.pct_change(periods=1).shift(-1)\n",
     "rolling_std = msft_returns.rolling(10).std()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because the first 9 values in the rolling_std are NaN due to the rolling window, there is no threshold to compare the first 9 returns to. So we must remove the first 9 returns as they are not labeled."
+   ]
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:239]: The above cell already states that `So we must remove the first 9 returns`. Perhaps this part should be changed.

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 5]
```diff
@@ -337,10 +336,82 @@
...
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "ax4 = bars4.value_counts().reindex([-1, 0, 1]).plot(kind='bar', x='label', y='frequency')\n",
+    "ax4.set_xlabel('Label')\n",
+    "ax4.set_ylabel('Frequency')"
+   ]
+  },
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:396]: Adding a small comparison (description) between Rolling standard deviation labels frequency bar plot and Threshold on market returns frequency bar plot would improve the readability of the Notebook. 

(Please, add it just before the `Labeling with standardization` part)

Also we can remove the `bars4.value_counts()` output as it's already presented better in the bar plot. (Same for other parts of this notebook)

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 6]
```diff
@@ -337,10 +336,82 @@
...
+   "source": [
+    "### Labeling with standardization "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Scaling each return by the mean and standard deviation is also a potential solution in getting around the non-homogeneaity of the data, as it could adjust for volatility differences between different bars. If standardization is desires, a rolling window must be given."
+   ]
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:409]: `non-homogeneaity` to `non-homogeneity`

`is desires` to `is desired`

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 7]
```diff
@@ -535,84 +606,61 @@
...
-     "output_type": "stream",
-     "text": [
-      "c:\\users\\ruifan\\mlfinlab\\mlfinlab\\labeling\\fixed_time_horizon.py:53: UserWarning: window is greater than the length of the Series. All labels will be NaN.\n",
-      "  warnings.warn('window is greater than the length of the Series. All labels will be NaN.', UserWarning)\n"
+     "ename": "AssertionError",
+     "evalue": "prices.index and threshold.index must match! If prices are resampled, the threshold index must match the resampled prices index.",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[1;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:632]: Perhaps showing errors would look better using the following structure (It is a shorter output that doesn't show your internal directory):

<pre>
try:
    fixed_time_horizon(prices=close[:10], threshold=threshold)
except Exception as exc:
    print(exc)
</pre>

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 8]
```diff
@@ -628,11 +676,11 @@
    "metadata": {},
    "source": [
     "This notebook presents the fixed horizon method, a simple method of labeling data for later use in machine learning algorithms. In this process:\n",
-    " - Forward rates of return for assets are calculated based on tick data, usually in the form of time bars.\n",
-    " - The forward return rate is compared to a threshold for the return. The threshold can be a constant, or a dynamic series corresponding to every timestamp in the data.\n",
+    " - Forward (lagged) returns for assets are calculated based on tick data, usually in the form of time bars. If applicable, the returns may be calculated from resampled prices.\n",
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:679]: In this notebook daily data is used, not tick.

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 9]
```diff
@@ -628,11 +676,11 @@
...
+    " - The forward return is compared to a threshold for the return. The threshold can be a constant, or a pd.Series corresponding to every timestamp in the data.\n",
     " - Each observation is compared to the threshold, and is labeled -1 if it is below -threshold, 0 if it is between -threshold and +threshold, and 1 if it is greater than +threshold.\n",
     "\n",
-    "In Dixon's paper, the data is labelled using this method. Subsequently a neural network is trained to to predict the label based on the data. A simple trading strategy in which a long position is taken if a ticker is classified as 1, and short if it's classified at -1, can then be applied."
+    "This method works with the prices input being pd.Series or pd.DataFrame. In Dixon's paper, the data is labeled using this method. Subsequently a neural network is trained to to predict the label based on the data. A simple trading strategy in which a long position is taken if a ticker is classified as 1, and short if it's classified at -1, can then be applied. "
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:683]: `to to` -> `to`

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb (hunk 4 above)
Comment (ruifanp) [reply Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:239]: removed the repetition

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb (hunk 7 above)
Comment (ruifanp) [reply Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:632]: yes much this is much better

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb (hunk 5 above)
Comment (ruifanp) [reply Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:396]: done

**Pull request merged**